### PR TITLE
fix: evaluation fails

### PR DESCRIPTION
**Pull request by kentakozuka**
fixes https://github.com/bucketeer-io/bucketeer/issues/968

### CODE REVIEW

File: evaluation/evaluation_test.go [hunk 1]
```diff
@@ -712,8 +770,11 @@ func TestEvaluateFeaturesByEvaluatedAt(t *testing.T) {
 				p.tag,
 			)
 			assert.Equal(t, p.expectedError, err)
+			if p.expectedError != nil {
+				return
+			}
 			assert.Equal(t, p.expectedEvals.UserEvaluations.ForceUpdate, actual.ForceUpdate)
-			assert.Equal(t, p.expectedEvals.UserEvaluations.ArchivedFeatureIds, actual.ArchivedFeatureIds)
+			assert.ElementsMatch(t, p.expectedEvals.UserEvaluations.ArchivedFeatureIds, actual.ArchivedFeatureIds)
```
Comment (kentakozuka) [inline evaluation/evaluation_test.go:777]: Fixes a flaky test.

File: evaluation/evaluation.go [hunk 2]
```diff
@@ -101,8 +102,10 @@ func (e *evaluator) EvaluateFeaturesByEvaluatedAt(
 	if len(updatedFeatures) == 0 {
 		return e.evaluate(fs, user, mapSegmentUsers, true, targetTag)
 	}
-	featuresHavePrerequisite := e.getFeaturesHavePrerequisite(fs)
-	evalTargets := e.getPrerequisiteUpwards(updatedFeatures, featuresHavePrerequisite)
+	evalTargets, err := e.getEvalFeatures(updatedFeatures, fs)
```
Comment (cre8ivejp) [inline evaluation/evaluation.go:105]: We need to add this before calling `evaluate` here, too.

https://github.com/bucketeer-io/bucketeer/blob/16f9c2fc693eba6749ce753cab8bf78a44b475c8/evaluation/evaluation.go#L70

Comment (cre8ivejp) [reply evaluation/evaluation.go:105]: The `GetEvaluation` API call this [getTargetFeatures](https://github.com/bucketeer-io/bucketeer/blob/16f9c2fc693eba6749ce753cab8bf78a44b475c8/pkg/gateway/api/api_grpc.go#L924-L934), which calls the `GetPrerequisiteDownwards`.

Comment (cre8ivejp) [reply evaluation/evaluation.go:105]: For this case, don't we also need to call the `getFeaturesDependedOnTargets`?

Comment (kentakozuka) [reply evaluation/evaluation.go:105]: It is not necessary to be added here because `fs` includes all features to be evaluated 👍

Comment (kentakozuka) [reply evaluation/evaluation.go:105]: I modified `GetPrerequisiteDownwards()` to call `getFeaturesDependedByTargets()`.

**Review by cre8ivejp — APPROVED** [review]
Thank you!

**Pull request merged**